api/apps/graphql/test/resolvers.test.js

```diff
@@ -0,0 +1,159 @@
+    resolvers.__set__('Author', { mongoFetch: (sinon.stub().yields(null, authors)) })
+  })
+
+  afterEach(() => {
+    server.close()
+  })
+
+  describe('articles', () => {
+    it('returns throws error when trying to view a draft without channel_id', () => {
+      const args = { published: false }
+      resolvers.articles.bind({}, args, {}, {}).should.throw()
+    })
+
+    it('returns throws an error when trying to view an unauthorized draft', () => {
+      const args = { published: false, channel_id: '123' }
+      resolvers.articles.bind({}, args, {}, {}).should.throw()
+    })
+
+    it('can view drafts', () => {
+      const args = { published: false, channel_id: '456' }
+      resolvers.articles({}, args, req, {})
+      .then(results => {
+        results.length.should.equal(1)
+        results[0].slug.should.equal('slug-1')
+      })
+    })
+
+    it('can view published articles', () => {
+      const args = { published: true }
+      resolvers.articles({}, args, req, {})
+      .then(results => {
+        results.length.should.equal(1)
+        results[0].slug.should.equal('slug-1')
+      })
+    })
+  })
+
+  describe('article', () => {
+    it('rejects with an error when no article is found', () => {
+      const args = {id: '123'}
+      resolvers.__set__('find', (sinon.stub().yields(null, null)))
+      resolvers.article({}, args, {}, {})
+      .catch(err => err.toString().should.containEql('Article not found.'))
+    })
+
+    it('rejects with an error when trying to view an unauthorized draft', () => {
+      const args = {id: '123'}
+      const newArticle = _.extend({}, article, {channel_id: '000', published: false})
+      resolvers.__set__('find', (sinon.stub().yields(null, newArticle)))
+      resolvers.article({}, args, {}, {})
+      .catch(err => err.toString().should.containEql('Must be a member of the channel'))
+    })
+
+    it('can view drafts', () => {
+      const args = {id: '123'}
+      const newArticle = _.extend({}, article, {published: false})
+      resolvers.__set__('find', (sinon.stub().yields(null, newArticle)))
+      resolvers.article({}, args, req, {})
+      .then(results => {
+        results.slug.should.equal('slug-2')
+      })
+    })
+
+    it('can view published articles', () => {
+      const args = {id: '123'}
+      resolvers.__set__('find', (sinon.stub().yields(null, article)))
+      resolvers.article({}, args, req, {})
+      .then(results => {
+        results.slug.should.equal('slug-2')
+      })
+    })
+  })
+
+  describe('channels', () => {
+    it('can find channels', () => {
+      resolvers.channels({}, {}, req, {})
+      .then((results) => {
+        results.length.should.equal(1)
+        results[0].name.should.equal('Editorial')
+        results[0].type.should.equal('editorial')
+      })
+    })
+  })
+
+  describe('curations', () => {
+    it('can find curations', () => {
+      resolvers.curations({}, {}, req, {})
```

Comment:
With this decaffeinated, you can use async await with these promises instead of `.then` e.g. `const results = await resovlers.curations({}, {}, req, {})`.